ray-operator/controllers/ray/common/pod_test.go

```diff
@@ -811,6 +820,20 @@ func TestDefaultInitContainer(t *testing.T) {
 
 	// Pass a deep copy of worker (*worker.DeepCopy()) to prevent "worker" from updating.
 	podTemplateSpec := DefaultWorkerPodTemplate(*cluster, *worker.DeepCopy(), podName, fqdnRayIP, "6379")
-	actualResult := len(podTemplateSpec.Spec.InitContainers)
-	assert.Equal(t, expectedResult, actualResult, "A default init container is expected to be added.")
+	numInitContainers := len(podTemplateSpec.Spec.InitContainers)
+	assert.Equal(t, expectedResult, numInitContainers, "A default init container is expected to be added.")
+
+	// This health-check init container requires certain environment variables to establish a secure connection
+	// with the Ray head using TLS authentication. Currently, we simply copied all environment variables from
+	// Ray container to the init container. This may be changed in the future.
+	healthCheckContainer := podTemplateSpec.Spec.InitContainers[numInitContainers-1]
+	rayContainer := worker.Template.Spec.Containers[getRayContainerIndex(worker.Template.Spec)]
+	assert.Equal(t, len(rayContainer.Env), len(healthCheckContainer.Env))
+	for _, env := range rayContainer.Env {
+		if env.Value != "" {
+			checkContainerEnv(t, healthCheckContainer, env.Name, env.Value)
+		} else {
+			checkContainerEnv(t, healthCheckContainer, env.Name, env.ValueFrom.FieldRef.FieldPath)
```

Comment:
I didn't really understand this if-else, could you explain it? Might be worth adding a comment

Reply:
The value of `EnvVar` can be set by either `EnvVar.Value` or `EnvVar.ValueFrom`. You can check [here](https://pkg.go.dev/k8s.io/api/core/v1#EnvVar) for more details. I will add some comments.

Reply:
Updated https://github.com/ray-project/kuberay/pull/989/commits/b6eb46fe312e504b98072d8cc50493e14515a684.